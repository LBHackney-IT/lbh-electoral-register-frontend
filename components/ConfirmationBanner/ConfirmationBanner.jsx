import PropTypes from 'prop-types';

const ConfirmationBanner = ({ num }) => {
  if (num === 0) {
    return '';
  } else if (num === 1) {
    return (
      <section className="lbh-page-announcement">
        <h3 className="lbh-page-announcement__title">
          Application successfully added to the electoral register
        </h3>
      </section>
    );
  } else {
    return (
      <section className="lbh-page-announcement">
        <h3 className="lbh-page-announcement__title">
          {num} applications successfully added to the electoral register
        </h3>
      </section>
    );
  }
};

ConfirmationBanner.propTypes = {
  num: PropTypes.number,
};

export default ConfirmationBanner;
