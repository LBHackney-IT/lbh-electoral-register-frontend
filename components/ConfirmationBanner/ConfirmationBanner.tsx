export interface Props {
  title: string;
  content?: string;
}

const ConfirmationBanner = ({ title, content }: Props): React.ReactElement => (
  <section className="lbh-page-announcement">
    <h4 className="lbh-page-announcement__title">{title}</h4>
    <div className="lbh-page-announcement__content">{content}</div>
  </section>
);

export default ConfirmationBanner;
