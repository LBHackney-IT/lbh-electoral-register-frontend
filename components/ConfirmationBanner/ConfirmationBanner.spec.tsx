import { render, screen } from '@testing-library/react';

import ConfirmationBanner from './ConfirmationBanner';

describe('ConfirmationBanner', () => {
  it('renders title', () => {
    render(<ConfirmationBanner title="This is the title"></ConfirmationBanner>);
    expect(screen.getByText('This is the title'));
  });

  it('renders content', () => {
    render(
      <ConfirmationBanner
        title="This is the title"
        content="This is the content"
      ></ConfirmationBanner>
    );
    expect(screen.getByText('This is the content'));
  });
});
