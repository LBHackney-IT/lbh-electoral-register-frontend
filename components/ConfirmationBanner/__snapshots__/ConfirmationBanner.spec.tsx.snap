// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationBanner should title and content properly 1`] = `
<DocumentFragment>
  <section
    class="lbh-page-announcement"
  >
    <h4
      class="lbh-page-announcement__title"
    >
      This is the title
    </h4>
    <div
      class="lbh-page-announcement__content"
    >
      This is the content
    </div>
  </section>
</DocumentFragment>
`;
